src/commands/google.ts

```diff
@@ -0,0 +1,98 @@
+import {CommandType} from "../types";
+import {logger, Config} from "@/config";
+
+import {
+  CacheType,
+  SlashCommandBuilder,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  Colors,
+} from "discord.js";
+
+// Interfaces for getting specific entries.
+interface SearchResult {
+  title: string;
+  link: string;
+}
+
+interface Results {
+  items: SearchResult[];
+}
+
+const googleModule: CommandType = {
+  data: new SlashCommandBuilder()
+    .setName("google")
+    .setDescription("Google something and return the top 10 results.")
+    .addStringOption((option) =>
+      option
+        .setName("query")
+        .setDescription("What you want to Google.")
+        .setRequired(true)
+    )
+    .addUserOption((option) =>
+      option
+        .setName("to")
+        .setDescription("The user you want to direct this to.")
+    ),
+  execute: async (interaction: ChatInputCommandInteraction<CacheType>) => {
+    const query = interaction.options.getString("query");
+    const toWhom = interaction.options.getUser("to");
+    try {
+      const embed = new EmbedBuilder()
+        .setColor(Colors.Blue)
+        .setAuthor({
+          name: `Requested by ${interaction.user.displayName}`,
+          iconURL: `${interaction.user?.avatarURL()}`,
+        })
+        .setTimestamp();
+
+      if (!Config.google_search_key || !Config.google_search_id) {
+        logger.info("No key or search engine ID.");
+        return;
+      }
+
+      const search_key = Config.google_search_key;
+      const search_id = Config.google_search_id;
+      const url = `https://www.googleapis.com/customsearch/v1?key=${search_key}&cx=${search_id}&q=${query}`;
+
+      let response: string = "";
+
+      const res = await fetch(url, {
+        method: "GET",
+        headers: {Accept: "*/*"},
```

Comment:
Its just occurred to me that since you're expecting JSON (see `res.json()` below, lol) you should specify that you will only accept `application/json`.